app/controllers/api_controller.rb

```diff
@@ -31,12 +31,40 @@ def current_facility_group
     current_user.facility.facility_group
   end
 
+  def current_sync_region
+    # This method selectively permits only FacilityGroup sync (via facility group ID)
+    # and block-level sync (via regions) and offers facility group as a safe fallback.
+    # Over time, the facility group ID support can be dropped and this method can
+    # allow other region types as well
+    return current_facility_group if requested_sync_region_id.blank?
+    return current_facility_group if requested_sync_region_id == current_facility_group.id
+    return current_block if block_level_sync?
```

Comment:
This looks nice. If I understand correctly,
* It's the FG if the client has no idea what a sync group is
* It's the FG if block-level sync is on, but the client still asks for an FG's data (because it saved the FG ID when block-level sync was off, and hasn't updated its facility info yet)